packages/wonder-blocks-announcer/src/index.ts

```diff
@@ -0,0 +1,334 @@
+// TODO: publish wonder-blocks-style package WB-1776
+// import {srOnly} from "../../wonder-blocks-style/src/styles/a11y";
+import type {
+    PolitenessLevel,
+    RegionDef,
+    RegionDictionary,
+    RegionFactory,
+} from "./announcer.types";
+
+const TIMEOUT_DELAY = 5000;
+let announcer: Announcer | null = null;
+
+export type SendMessageProps = {
+    message: string;
+    level?: PolitenessLevel;
+    removalDelay?: number;
+};
+
+/**
+ * Public API method to send screen reader messages.
+ * @param {string} message The message to send.
+ * @param {PolitenessLevel} level Polite or assertive announcements
+ * @param {number} removalDelay Optional duration to remove a message after sending. Defaults to 5000ms.
+ * @returns {string} IDREF for targeted live region element
+ */
+export function sendMessage({
+    message,
+    level = "polite", // TODO: decide whether to allow other roles, i.e. role=`timer`
+    removalDelay,
+}: SendMessageProps): string {
+    announcer = Announcer.getInstance();
+
+    if (typeof jest === "undefined") {
+        setTimeout(() => {
+            return announcer?.announce(message, level, removalDelay);
+        }, 100);
+    } else {
+        // If we are in a test environment, announce without waiting
+        return announcer.announce(message, level, removalDelay);
+    }
```

Comment:
**suggestion:** I'd strongly recommend not using test checks in the prod code. Instead, you could probably use advanceTimersByTime in during test setup.

e.g.
```
// Act
do something to trigger the announcer
jest.advanceTimersByTime(100);
```

For reference: https://jestjs.io/docs/timer-mocks#advance-timers-by-time

You can see some existing examples in the codebase.

Reply:
This is a good callout! I tried using the Jest functionality but it presented two challenges: 1) it became overly verbose to call before every method. and 2) the tests failed as the setTimeout was still being called.

Instead, I'm proposing a `timeoutDelay` parameter to adjust the announcement timeout duration (in addition to the `removalDelay`). That way the code can check if it's set to 0 and skip the initial timeout that way, rather than referencing Jest.

Reply:
cool! dependency injection FTW! (kinda)